chain-core/src/common/merkle_tree.rs

```diff
@@ -113,13 +114,15 @@ impl<T> Tree<T> {
 
 /// Sibling's hash
 #[derive(Debug, PartialEq, Eq, Clone, Encode, Decode)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub enum Sibling {
     Left(H256),
     Right(H256),
 }
 
 /// Merkle path for inclusion proof
 #[derive(Debug, PartialEq, Eq, Clone, Encode, Decode)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
```

Comment:
`Serialize` and `Deserialize` shouldn't be added here.

Reply:
without these  compilation failed,
but i'll try agian